docs/build/getting-started/oracles.md

```diff
@@ -0,0 +1,129 @@
+
+:::tip
+
+You can learn more about Supra Oracles in their [official documentation](https://supraoracles.com/docs/overview).
+
+:::
+
+### Pyth
+
+The Pyth Network delivers real-time financial market data across multiple blockchains and can be used in off-chain applications.
+Pyth gets contributions from over 90 publishers, including several exchanges. It offers comprehensive and accurate price
+feeds for asset classes such as US equities, commodities, and cryptocurrencies. Its price feeds get updates
+multiple times per second, ensuring they are current and reliable.
+
+:::tip
+
+You can learn more about Pyth Oracles in their [official documentation](https://docs.pyth.network/documentation).
+
+:::
+
+#### Example Code
+
+You can use the following example code to get data from Pyth Oracles in Shimmer EVM:
+
+:::note 
+
+The following example uses [`ether.js` v4](https://www.npmjs.com/package/ethers/v/4.0.43).
+
+:::
+
+```typescript
+import {ethers} from "ethers";
+import fetch from "node-fetch";
+
+// Provider
+const providerURL = "https://json-rpc.evm.shimmer.network"
+const provider = new ethers.providers.JsonRpcProvider(providerURL);
+
+//  Pyth Oracle contract on ShimmerEVM Mainnet
+const contractAddress = '0xA2aa501b19aff244D90cc15a4Cf739D2725B5729';
+
+// Add Price Feed Ids obtained from https://pyth.network/developers/price-feed-ids#pyth-evm-stable
+const priceFeedIds = [
+    //SMR/USD
+    '0xaf5b9ac426ae79591fde6816bc3f043b5e06d5e442f52112f76249320df22449',
+    //BTC/USD
+    '0xe62df6c8b4a85fe1a67db44dc12de5db330f7ac66b72dc658afedf0f4a415b43'
+]
+
+async function getPrice() {
+  // Replace 'IPythJSONURL' with the actual URL where the IPyth ABI JSON is hosted
+  const pythAbiUrl = 'https://raw.githubusercontent.com/pyth-network/pyth-sdk-solidity/main/abis/IPyth.json';
+  const pythAbi = await fetch(pythAbiUrl).then((response) => response.json());
+
+  const pythContract = new ethers.Contract(contractAddress, pythAbi, provider);
+
+   try {
+    for (const priceFeedId of priceFeedIds) {
+      const priceData = await pythContract.getPrice(priceFeedId);
+      const price = Number(priceData.price) * 10 ** (Number(priceData.expo));
+
+      console.log(`Price for ${priceFeedId}:`);
+      console.log('Price:', price);
+      console.log('Confidence Interval:', priceData.conf.toString());
+      // Convert publishTime to a formatted date
+      const publishTimeDate = new Date(priceData.publishTime.toNumber() * 1000);
+      console.log('Publish Time:', publishTimeDate.toUTCString());
+      console.log('\n');
+    }
+  } catch (error) {
+    console.error('Error:', error.message);
+  }
+}
+
+async function main() {
+  try {
+    await getPrice();
+  } catch (error) {
+    console.error('Main Error:', error.message);
+  }
+}
+
+main();
+```
```

Comment:
```suggestion

```typescript
import {ethers} from "ethers";
import fetch from "node-fetch";

// Provider
const providerURL = "https://json-rpc.evm.shimmer.network"
const provider = new ethers.JsonRpcProvider(providerURL);

//  Pyth Oracle contract on ShimmerEVM Mainnet
const contractAddress = '0xA2aa501b19aff244D90cc15a4Cf739D2725B5729';

// Add Price Feed Ids obtained from https://pyth.network/developers/price-feed-ids#pyth-evm-stable
const priceFeedIds = [
    //SMR/USD
    '0xaf5b9ac426ae79591fde6816bc3f043b5e06d5e442f52112f76249320df22449',
    //BTC/USD
    '0xe62df6c8b4a85fe1a67db44dc12de5db330f7ac66b72dc658afedf0f4a415b43'
]

async function getPrice() {
  // Replace 'IPythJSONURL' with the actual URL where the IPyth ABI JSON is hosted
  const pythAbiUrl = 'https://raw.githubusercontent.com/pyth-network/pyth-sdk-solidity/main/abis/IPyth.json';
  const pythAbi = await fetch(pythAbiUrl).then((response) => response.json());

  const pythContract = new ethers.Contract(contractAddress, pythAbi, provider);

   try {
    for (const priceFeedId of priceFeedIds) {
      const priceData = await pythContract.getPrice(priceFeedId);
      const price = Number(priceData.price) * 10 ** (Number(priceData.expo));

      console.log(`Price for ${priceFeedId}:`);
      console.log('Price:', price);
      console.log('Confidence Interval:', priceData.conf.toString());
      // Convert publishTime to a formatted date
      const publishTimeDate = new Date(number(priceData.publishTime) * 1000);
      console.log('Publish Time:', publishTimeDate.toUTCString());
      console.log('\n');
    }
  } catch (error) {
    console.error('Error:', error.message);
  }
}

async function main() {
  try {
    await getPrice();
  } catch (error) {
    console.error('Main Error:', error.message);
  }
}

main();
```
```

Reply:
This works for me with ethers v6